dll/win32/shell32/shlexec.cpp

```diff
@@ -1042,6 +1042,18 @@ static unsigned dde_connect(const WCHAR* key, const WCHAR* start, WCHAR* ddeexec
             TRACE("Couldn't launch\n");
             goto error;
         }
+        /* if ddeexec is NULL, then we just need to exit here */
+        if (ddeexec == NULL)
+        {
+            TRACE("Exiting because ddeexec is NULL. ret=33.\n");
+            return 33; /* FIXME see SHELL_FindExecutable() */
```

Comment:
Whyt is this magic number? Can we have a symbolic constant?

Reply:
This number and the "return 33;" is a copy from previously used code in line numbers 677, 730, and 802 as programmed by Ged Murphy (@gedmurphy) according to "Blame". I would be happy to replace this by a symbolic constant if you supply a name that you like. (GREATER_THAN_32_AND_OK ?)

Reply:
@tkreuzer, I have done more research and from my testing Windows Server 2003 has returned the value "42" in all cases for me on a successful ShellExecuteW. Therefore, it seems that we could add into shellapi.h a statement of "#define SE_ERR_NONE 42" and change the "return 33;" statements to "return SE_ERROR_NONE;" Comments?

Reply:
42 is obviously the answer to everything 😄 If this is not documented anywhere, I'm fine with keeping it like this, but preferably with a comment pointing to the docs and that the actual value is based on tests. Would be great to add that test somewhere, too.

Reply:
New https://github.com/reactos/reactos/pull/1854 created to demonstrate the return values. I suspect that the 42 is a nod to Douglas Adams who wrote "Hitchhiker's Guide to the Galaxy" as you suggest. I will update this PR with comments later as you suggest. Thanks.